testing/src/scenario/state.py

```diff
@@ -1975,6 +1975,10 @@ def autoload(charm_type: type[CharmBase]) -> _CharmSpec[CharmType]:
             # try to load using legacy metadata.yaml/actions.yaml/config.yaml files
             meta, config, actions = _CharmSpec._load_metadata_legacy(charm_root)
 
+        # TODO: ideally, we would look for ConfigBase classes in the charm
+        # module and autoload from there at this point. Leaving this until the
+        # conversation about if & how the generation is done is resolved.
+
```

Comment:
My feeling at the moment is that this should be left for future work, rather than baking anything in to Scenario at this time. There is a simple example of using this functionality in tests for both config and actions in the Scenario tests - I think those show that it's not too arduous to use at the moment, without anything else.